src/Microsoft.AspNetCore.Routing/Matching/DfaMatcherBuilder.cs

```diff
@@ -388,10 +388,15 @@ public override Matcher Build()
                 var policyBuilder = new PolicyJumpTableBuilder(node.NodeBuilder);
                 tableBuilders[currentStateIndex] = (pathBuilder, policyBuilder);
 
+                var entries = new PolicyJumpTableEdge[node.PolicyEdges.Count];
+
+                var index = 0;
                 foreach (var kvp in node.PolicyEdges)
                 {
-                    policyBuilder.AddEntry(kvp.Key, Transition(kvp.Value));
+                    entries[index++] = new PolicyJumpTableEdge(kvp.Key, Transition(kvp.Value));
                 }
+
+                policyBuilder.AddEntries(entries);
```

Comment:
Consider using the node builder directly (i.e. no longer use the **PolciyJumpTableBuilder**)?